analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ExceptionStateTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn
+{
+    [TestClass]
+    public partial class ExceptionStateTest
+    {
+        [TestMethod]
+        public void Constructor_Null_Throws() =>
+            ((Func<ExceptionState>)(() => new ExceptionState(null))).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("type");
+
+        [TestMethod]
+        public void ToString_Unknown() =>
+            ExceptionState.UnknownException.ToString().Should().Be("Unknown");
+
+        [DataTestMethod]
+        [DataRow("System.Exception", "Exception")]
+        [DataRow("System.ArgumentNullException", "ArgumentNullException")]
+        [DataRow("System.IO.IOException", "IOException")]
+        public void ToString_Known(string typeName, string expected) =>
+            new ExceptionState(TestHelper.CompileCS(string.Empty).Model.Compilation.GetTypeByMetadataName(typeName)).ToString().Should().Be(expected);
```

Comment:
To string will be used only for debugging? We could have wrong results when comparing exceptions with the same name from different namespaces.

Reply:
It should be used only for debugging and debugging serialization (in ProgramState).

Branching decisions must be made on the `Type` property.

Reply:
Discussed offline and it's preferred to have it as a string.